### DISCUSSION TITLE
useing $args in the resolve cloture of a type item

### DISCUSSION THREAD

**Discussion by mehrabx** [rebing/graphql-laravel#977] [Q&A]
### Versions:
- graphql-laravel Version: 8.2
- Laravel Version: 9.25
- PHP Version: 8.0

### Description:

i am trying to use $args in the resolve cloture of a type item but it is always null.
If it's about using ``SelectFields``, I used it.

```
  'example_item_of_the_type' => [
      'type'        => Type::string(),
      'description' => 'example description',
      'resolve'     => function ($root, array $args) {
          dd($args); // it shows null
      },
  ],
```

**Comment by mfn** [comment]
You will only have `$args` in there, if you provided them on this field on the query -> how does your query look like?

**Comment by mehrabx** [comment]
How can I access to client request arguments in this resolve cloture?

**Reply by mfn** [reply]
> client request arguments

What do you mean specifically with this?

**Reply by mehrabx** [reply]
Suppose a client sends a query like the following:
```
query {
    Newsletters
    (
        sort_by : "latest" 
        get_title_from_cache : true
    )
    {
      id
      title
    }
}
```
And a response of the following graphql type is going to be returned: 
```
return [
            'id'      => [
                'type'        => Type::string(),
            ],
            'title'   => [
                'type'        => Type::string(),
                'resolve'     => function ($root, array $args) {
                    if($args['get_title_from_cache']){ 🔴 
                        //get title from cache
                    }
                },
            ],
      ]
```
The red dot is where I need the user query argument and it is not available

**Reply by mfn** [reply]
Right, because that would require you to provide the args on the `title`.
```graphql
query {
    Newsletters
    (
        sort_by : "latest" 
        get_title_from_cache : true
    )
    {
      id
      title(get_title_from_cache:true) 👈
    }
}
```
The earlier versions of graphql-laravel actually worked like you would have expected this, but when once then used "local arguments" (not sure if this is the right term) _and_ global args on the query/mutation, this causes argument shadowing/confusion and bugs.

I'm not sure there is another way, it's possible that the resolver for `title` receives `ResolveInfo` and maybe you can traverse your way through this datastructure and get the parent/top level args ou of it, but I'm afraid that's uncharted territory for me so not sure I can help more than that.
